### PR TITLE
feat(web): multi channel provider config - integrations list modal

### PR DESCRIPTION
**Pull request by LetItRock**
### What change does this PR introduce?

**Don't worry about the number of file changes, most of them include import updates.**

The new Integrations List Modal:
- changed the existing sidebar components for selecting, creating, and updating the integration to be free of hardcoded routes logic
- created the missing page wrappers for the sidebars
- created the new Integrations List Modal component that reuses the existing components like integrations list and sidebars
- fixed types that were duplicated
- removed the validation of credentials on the BE that I added some time ago because some of the integrations do not require any credentials like `MSTeams`

### Why was this change needed?

Multi Provider Config PRD.

### Other information (Screenshots)

Screenshot 2023-07-13 at 14 49 29

Screenshot 2023-07-13 at 14 01 08

Screenshot 2023-07-13 at 14 01 17

Screenshot 2023-07-13 at 14 01 26

Screenshot 2023-07-13 at 14 01 35

Screenshot 2023-07-13 at 14 01 43

https://github.com/novuhq/novu/assets/2607232/9d00566c-a6a5-4b62-b64c-9e6dcb77d208

https://github.com/novuhq/novu/assets/2607232/bfd89313-3d8b-43cc-af81-ac17cfd9d7bd

https://github.com/novuhq/novu/assets/2607232/6bc9daff-8d9e-4a88-9d95-f739e8c6a5fb

https://github.com/novuhq/novu/assets/2607232/25be4062-e049-4a91-b27e-ccdadd78cdc8

https://github.com/novuhq/novu/assets/2607232/9675c690-65a5-4d67-adb6-0d01c2e19c0e

### CODE REVIEW

File: apps/web/src/pages/integrations/UpdateProviderPage.tsx [hunk 1]
```diff
@@ -1,274 +1,15 @@
...
-}
-
-enum SidebarStateEnum {
-  NORMAL = 'normal',
-  EXPANDED = 'expanded',
-}
+import { UpdateProviderSidebar } from './components/multi-provider/UpdateProviderSidebar';
 
 export function UpdateProviderPage() {
-  const { environments, isLoading: areEnvironmentsLoading } = useFetchEnvironments();
```
Comment (LetItRock) [inline apps/web/src/pages/integrations/UpdateProviderPage.tsx:50]: all this code moved to the `UpdateProviderSidebar`

File: apps/web/src/pages/integrations/components/multi-provider/CreateProviderInstanceSidebar.tsx [hunk 2]
```diff
@@ -16,21 +15,31 @@ import { errorMessage, successMessage } from '../../../../utils/notifications';
...
-  provider,
   onGoBack,
+  onIntegrationCreated,
 }: {
+  isOpened: boolean;
+  channel?: string;
+  providerId?: string;
   onClose: () => void;
   onGoBack: () => void;
-  provider: IProviderConfig;
```
Comment (LetItRock) [inline apps/web/src/pages/integrations/components/multi-provider/CreateProviderInstanceSidebar.tsx:28]: removed this prop and added `channel, providerId` to make it more reusable

File: apps/web/src/pages/templates/components/ListProviders.tsx [hunk 3]
```diff
@@ -3,16 +3,18 @@ import { ChannelTypeEnum } from '@novu/shared';
...
 import { ChannelTitle } from './ChannelTitle';
 import { LackIntegrationError } from './LackIntegrationError';
 
 export const ListProviders = ({
+  channel,
   providers,
   setConfigureChannel,
   setProvider,
 }: {
+  channel: ChannelTypeEnum;
```
Comment (LetItRock) [inline apps/web/src/pages/templates/components/ListProviders.tsx:17]: changed to pass the channel directly than having the logic to take the first el and take the channel from there, which was breaking

**Review by ainouzgali — APPROVED** [review]
Great work! Looks good

**Pull request merged**